### ISSUE TITLE
user.progress(course) fails when the course has no progress at all.

### ISSUE THREAD

**Issue by garywu** [tylucaskelley/udacity-api-python#3]
```
Traceback (most recent call last):
  File "cli.py", line 8, in <module>
    prog = user.progress(course)
  File "C:\Users\Dev\code\source\udacity-api-python\udacity\user.py", line 301, in progress
    return get_course_progress(course_data)
  File "C:\Users\Dev\code\source\udacity-api-python\udacity\user.py", line 252, in get_course_progress
    most_recent_page = max(last_interactions, key=lambda i: i['time'])
ValueError: max() arg is an empty sequence
```

This is the json returned by the server in 
user.py line 248:
course_progress = json.loads(req.text[4:])['nodestates'][0]
last_interactions is empty thus cause above error.

```javascript
{u'account_ref': {u'key': u'4141508737', u'ref': u'Account'},
 u'bin_codes': [],
 u'completed_descendant_node_refs': [],
 u'completed_duration': 0,
 u'completed_morsel_count': 0,
 u'completed_quiz_count': 0,
 u'last_interactions': [],
 u'last_submission_data': None,
 u'marked_completed': False,
 u'mastery_points_summary': {},
 u'node_completed': False,
 u'node_ref': {u'key': u'ud585', u'ref': u'Node'},
 u'score': None}
```

**Comment by tylucaskelley** [comment]
Yay for edge cases :) will fix this and add new tests when I get a chance! (should be sometime this week)

**Comment by tylucaskelley** [comment]
fixed in commit 647dd14

**Comment by tylucaskelley** [comment]
published v1.1.1 with fix
